src/test/java/com/cegeka/academy/repository/InterestTest.java

```diff
@@ -61,4 +67,33 @@ public void testFindByDescription(){
         assertThat(interestRepository.findByDescription("description1").get(0)).isEqualTo(interestRepository.findAll().get(0));
     }
 
+    @Test
+    public void testAddUserInterest(){
+        Interest interest1  = new Interest();
+        interest1.setName("interest1");
+        interest1.setDescription("description1");
+
+        User user = new User();
+        user.setEmail("gigi.gogaie@gmail.com");
+        user.setFirstName("Gogaie");
+        user.setLastName("Momaie");
+        user.setPassword("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+        user.setLogin("gigel@purcel.com");
+
+        Set userSet = new HashSet<>();
+        Set interestSet = new HashSet<>();
+
+        userSet.add(user);
+        interestSet.add(interest1);
+
+        user.setUserInterests(interestSet);
+        interest1.setInterestUsers(userSet);
+
+        interestRepository.save(interest1);
+        userRepository.save(user);
+
+        assertThat(userRepository.findAll().get(4).getUserInterests().equals(interestSet)).isTrue();
```

Comment:
De ce ai get(4)? Daca faci cu @Before si @After scapi de aceasta problema.

Reply:
Am incercat sa salvez intr-o lista si sa sterg din baza de date in "before" inregistrarile dar primeam o exceptie. Nu ma lasa sa le sterg nici cu deleteAll() si nici cu deleteAll(Iterable...) daca ii specificam exact inregistrarile.